substrate/frame/staking/src/pallet/mod.rs

```diff
@@ -336,7 +336,7 @@ pub mod pallet {
 	/// TWOX-NOTE: SAFE since `AccountId` is a secure hash.
 	#[pallet::storage]
 	pub type Payee<T: Config> =
-		StorageMap<_, Twox64Concat, T::AccountId, RewardDestination<T::AccountId>, ValueQuery>;
+		StorageMap<_, Twox64Concat, T::AccountId, RewardDestination<T::AccountId>, OptionQuery>;
```

Comment:
Having this be an `OptionQuery` now feels bad because, as you pointed out, if there is a ledger, there should be a reward destination. It's also true that a reward destination without a ledger doesn't make sense. So because there is this 1:1 mapping here, it would make sense to put the reward destination in the same place as the ledger or the rest of the information. This is in order to avoid having to do things like

```
let ledger = ledger().ok_or(...)?;
let reward_destination = reward_destination().expect("must be because of line above; qed");
```

An error instead of the `expect` doesn't look much better to me either.

Modifying the `StakingLedger` structure requires a migration and I'm not sure if it's worth it. I'm curious if there is another cleaner way to fix this logical inconsistency with the always `Some` `Option`.

Reply:
> Having this be an OptionQuery now feels bad because, as you pointed out, if there is a ledger, there should be a reward destination

I always pictured a `None` payee would be the equivalent of `RewardDestination::None`, with that variant no longer being needed if there was an `OptionQuery`. But ultimately this may not make much difference storage-wise, as i imagine just about all stakers will want rewards going somewhere.

Reply:
I do like that `Default::default()` is no longer being used though from a readability perspective.

Reply:
I don't see a nice and easy way to fix this right now, so I'm fine with the current approach.